Obsdocs 1803 (#94636)
openshift/openshift-docs: scheduling_resources/scheduling-logging-resources.adoc

> include::_attributes/common-attributes.adoc[]

Reviewer: This needs to follow our style **[guide](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#assembly-file-metadata)**

Author: done

:_newdoc-version: 2.18.4
:_template-generated: 2025-06-24
:_mod-docs-content-type: ASSEMBLY

[id="scheduling-logging-resources"]
= Scheduling logging resources

toc::[]

:context: scheduling-logging-resources

> You can control logging resources scheduling by defining node selectors, taints and tolerations, and affinity and anti-affinity configurations. 

Reviewer: ```suggestion
You can schedule logging resources by defining node selectors, taints and tolerations, and affinity and anti-affinity configurations. 
```

> Node selector:: A node selector specifies a map of key/value pairs that are defined using custom labels on nodes and selectors specified in pods. For the pod to be eligible to run on a node, the pod must have the same key/value node selector as the label on the node.

Reviewer: ```suggestion
Node selector:: A node selector specifies a map of key/value pairs that are defined using custom labels on nodes and selectors specified in pods. For the pod to be eligible to run on a node, the pod must have the same key/value pair as the label on the node.
```

Author: done

> Taints and toleration:: Taints and tolerations allow the node to control which pods should, or should not, be scheduled on them.

Reviewer: ```suggestion
Taints and toleration:: Taints and tolerations control which pods should, or should not, be scheduled on nodes.
```

> Affinity and anti-affinity:: Pod affinity and pod anti-affinity allow you to constrain which nodes your pod is eligible to be scheduled on based on the key-value labels on other pods.

Reviewer: This is another way you could reword line 18.

> Affinity and anti-affinity:: Pod affinity and pod anti-affinity allow you to constrain which nodes your pod is eligible to be scheduled on based on the key-value labels on other pods.

Reviewer: ```suggestion
Affinity and anti-affinity:: Pod affinity and pod anti-affinity allow you to constrain which nodes your pod is eligible to be scheduled on based on the key/value labels on other pods.
```

> [IMPORTANT]

Reviewer: It isn't technically wrong to callout this information in an IMPORTANT admonition. However, this isn't just a single consideration. With that in mind, I would move these into their own module or add them to the assembly with another heading `Considerations when using node selectors and node affinity rules`.

Author: removed the text about nodeSelectorTerms, so this is a single consideration now.

====
If you are using node selectors and node affinity in the same pod configuration, the following rules control pod placement onto nodes:

> * If you configure both `nodeSelector` and `nodeAffinity`, both conditions must be satisfied for the pod to be scheduled onto a candidate node.

Reviewer: ```suggestion
* If you configure both `nodeSelector` and `nodeAffinity` fields, the conditions of both fields must be met for the pod to be scheduled onto a candidate node.
```

> * If you specify multiple `nodeSelectorTerms` associated with `nodeAffinity` types, then the pod can be scheduled onto a node if one of the `nodeSelectorTerms` is satisfied.

Reviewer: ```suggestion
* If you specify multiple `nodeSelectorTerms` associated with `nodeAffinity` types, then the pod can be scheduled onto a node if at least one of the `nodeSelectorTerms` is satisfied.
```

Reviewer: Same comment about `nodeSelectorTerms` applies here.

Author: fixed

> * If you specify multiple `matchExpressions` associated with `nodeSelectorTerms`, then the pod can be scheduled onto a node only if all `matchExpressions` are satisfied.

Reviewer: ```suggestion
* If you specify multiple `matchExpressions` types associated with `nodeSelectorTerms`, then the pod can be scheduled onto a node only if all `matchExpressions` are satisfied.
```

Reviewer: I don't know why we are mentioning `nodeSelectorTerms` because the example does not use them, nor does the module define what they are so the user has no way of knowing how to use these or what they are.
